apps/tests/aws-runtime/test/test-service.ts

```diff
@@ -793,6 +841,22 @@ export const entityWorkflow = workflow(
       entitySignal2.expectSignal(),
     ]);
 
+    if (
+      orderError ||
+      // wait 30 seconds for the order value to match the final value.
+      !(await condition(
+        { timeout: duration(300, "seconds") },
+        () => finalValue?.n === orderValue
+      ))
+    ) {
+      if (orderError) {
+        throw new Error(orderError);
+      }
+      throw new Error(
+        `Order handler never received the final value! have: ${orderValue} expected: ${finalValue?.n}`
+      );
```

Comment:
I don't understand

Reply:
The test is testing stream handling order by sending signals to the workflow. The event/singal order will also be maintained. If the workflow sees `n` values out of order, it should fail.

`orderError` - signal handlers cannot throw to the caller, so I put the error into a variable and check it at the same time I check the order.